packages/commerce-sdk-react/src/hooks/ShopperProducts/query.ts

```diff
@@ -7,22 +7,59 @@
 import {ApiClients, Argument, DataType} from '../types'
 import {useAsync} from '../useAsync'
 import useCommerceApi from '../useCommerceApi'
-import {UseQueryResult} from '@tanstack/react-query'
+import {UseQueryOptions, UseQueryResult} from '@tanstack/react-query'
 
 type Client = ApiClients['shopperProducts']
-
 /**
  * A hook for `ShopperProducts#getProducts`.
  * Allows access to multiple products by a single request. Only products that are online and assigned to a site catalog are returned. The maximum number of productIDs that can be requested are 24. Along with product details, the availability, images, price, promotions, and variations for the valid products will be included, as appropriate.
  * @see {@link https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-products?meta=getProducts} for more information about the API endpoint.
  * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shopperproducts.shopperproducts-1.html#getproducts} for more information on the parameters and returned data type.
  * @returns An object describing the state of the request.
  */
-export const useProducts = (
-    arg: Argument<Client['getProducts']>
-): UseQueryResult<DataType<Client['getProducts']>, Error> => {
+type UseProductsParameters = NonNullable<Argument<Client['getProducts']>>['parameters']
+type UseProductsHeaders = NonNullable<Argument<Client['getProducts']>>['headers']
+function useProducts(
+    parameters: UseProductsParameters,
+    options?: {
+        headers: UseProductsHeaders
+        rawResponse: false
+        queryOptions: UseQueryOptions<DataType<Client['getProducts']> | Response, Error>
+    }
+): UseQueryResult<DataType<Client['getProducts']>, Error>
+function useProducts(
+    parameters: UseProductsParameters,
+    options?: {
+        headers: UseProductsHeaders
+        rawResponse: true
+        queryOptions?: UseQueryOptions<DataType<Client['getProducts']> | Response, Error>
+    }
+): UseQueryResult<Response, Error>
+function useProducts(
+    parameters: UseProductsParameters,
+    options?: {
+        headers: UseProductsHeaders
+        rawResponse: boolean
+        queryOptions?: UseQueryOptions<DataType<Client['getProducts']> | Response, Error>
+    }
+): UseQueryResult<DataType<Client['getProducts']> | Response, Error>
+function useProducts(
+    parameters: UseProductsParameters,
+    options?: {
+        headers: UseProductsHeaders
+        rawResponse: boolean
+        queryOptions?: UseQueryOptions<DataType<Client['getProducts']> | Response, Error>
+    }
+) {
+    if (!parameters) {
+        throw new Error('parameters are required')
+    }
     const {shopperProducts: client} = useCommerceApi()
-    return useAsync(['products', arg], () => client.getProducts(arg))
+    return useAsync(
+        ['products', parameters],
```

Comment:
About the 1st item in this query key: currently we have `products`, `product`, `categories`, and `category`. It looks like these came from the methods names in the isomorphic sdk.

Have we considered the 1st items of the query keys to be similar to the **API endpoints**? In other words, what if they're grouped as `products` and `categories` (and no singular `product`, `category`). Because the endpoint urls do not have the singulars.

I think there are benefits to doing it that way. But I can't think of any example right now 😅